torch/_inductor/codegen/cpp.py

```diff
@@ -784,19 +788,33 @@ def load(self, name: str, index: sympy.Expr):
         expanded_index = sympy.expand(index)
         new_index = self.transform_index(index)
 
-        if expanded_index == new_index:
-            line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
-        else:
-            if V.graph.get_dtype(name) in [torch.bool, torch.uint8]:
+        def load_help(load_index, nelements):
+            if V.graph.get_dtype(name) != torch.float:
```

Comment:
For non float type, I'm wondering whether we would be more conservative and only do explicit convert to float for scalar  load here? Here the assumption is that it is safe to convert a double scalar to float since this scalar would be down-casted to the tensor type of float before computing according to the type promotion rules of PyTorch. But it is not true for non-scalar cases where the data type would be promoted to the most precise one.

Reply:
Also note that we have to generalize the data type a bit in the future to support bfloat16/half etc. and not hard-code the cast to float here. Not necessarily in this PR.

Reply:
For vector check, we only support float dtype except for load and to_dtype, I think it  is safe to convert other type to float, because we have made sure all operators need float inputs under the current for the block.